metrics.go

```diff
@@ -0,0 +1,266 @@
+package frankenphp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"path/filepath"
+	"regexp"
+	"sync"
+	"time"
+)
+
+type Metrics interface {
+	// StartWorker Collects started workers
+	StartWorker(name string)
+	// StopWorker Collects stopped workers
+	StopWorker(name string)
+	// TotalWorkers Collects expected workers
+	TotalWorkers(name string, num int)
+	// TotalThreads Collects total threads
+	TotalThreads(num int)
+	// StartRequest Collects started requests
+	StartRequest()
+	// StopRequest Collects stopped requests
+	StopRequest()
+	// StopWorkerRequest Collects stopped worker requests
+	StopWorkerRequest(name string, duration time.Duration)
+	StartWorkerRequest(name string)
+	RenameWorker(oldName, newName string)
+	Shutdown()
+}
+
+type nullMetrics struct{}
```

Comment:
I'm not totally against this (I know the rationale behind the underlying pattern) but it feels a bit overkill to avoid 2 or 3 if 😅

Reply:
I assume we will be adding more interesting metrics in the future (maybe collecting worker memory usage, etc) and from experience, I found it error-prone by requiring `if metrics != nil` all over the place—not to mention how hard it was to scan the code.